services/motion/builtin/testutils.go

```diff
@@ -0,0 +1,542 @@
+	deps[motors[rightMotorName].Name()] = motors[rightMotorName]
+	deps[fakeBase.Name()] = fakeBase
+	deps[sens.Name()] = sens
+
+	return deps
+}
+
+func createWheeledBase(
+	t *testing.T,
+	ctx context.Context,
+	logger logging.Logger,
+	motors map[string]motor.Motor,
+	geomSize float64,
+) base.Base {
+	t.Helper()
+
+	reg, ok := resource.LookupRegistration(base.API, wheeled.Model)
+	test.That(t, ok, test.ShouldBeTrue)
+
+	wheeledConf := &wheeled.Config{
+		WidthMM:              200,
+		WheelCircumferenceMM: 2000,
+		Left:                 []string{leftMotorName},
+		Right:                []string{rightMotorName},
+	}
+
+	baseConf := resource.Config{
+		Name:                baseName,
+		API:                 base.API,
+		Frame:               &referenceframe.LinkConfig{Geometry: &spatialmath.GeometryConfig{R: geomSize}},
+		ConvertedAttributes: wheeledConf,
+	}
+	deps := resource.Dependencies{}
+	deps[motors[leftMotorName].Name()] = motors[leftMotorName]
+	deps[motors[rightMotorName].Name()] = motors[rightMotorName]
+
+	wBase, err := reg.Constructor(ctx, deps, baseConf, logger)
+	test.That(t, err, test.ShouldBeNil)
+	return wBase.(base.Base)
+}
+
+func createEncoders(t *testing.T, ctx context.Context, logger logging.Logger) map[string]encoder.Encoder {
+	t.Helper()
+	leftconf := resource.Config{
+		Name:                leftMotorName + "_encoder",
+		API:                 encoder.API,
+		ConvertedAttributes: &fakeencoder.Config{UpdateRate: int64(updateRate)},
+	}
+	rightconf := resource.Config{
+		Name:                rightMotorName + "_encoder",
+		API:                 encoder.API,
+		ConvertedAttributes: &fakeencoder.Config{UpdateRate: int64(updateRate)},
+	}
+	left, err := fakeencoder.NewEncoder(ctx, leftconf, logger)
+	test.That(t, err, test.ShouldBeNil)
+	right, err := fakeencoder.NewEncoder(ctx, rightconf, logger)
+	test.That(t, err, test.ShouldBeNil)
+	return map[string]encoder.Encoder{leftMotorName: left, rightMotorName: right}
+}
+
+func createMotors(
+	t *testing.T,
+	ctx context.Context,
+	logger logging.Logger,
+	encoders map[string]encoder.Encoder,
+) map[string]motor.Motor {
+	t.Helper()
+	motorMap := map[string]motor.Motor{}
+	for name, encoder := range encoders {
+		conf := resource.Config{
+			Name:                name,
+			API:                 motor.API,
+			ConvertedAttributes: &fakemotor.Config{Encoder: name + "_encoder", TicksPerRotation: 4000},
+		}
+
+		thisMotor, err := fakemotor.NewMotor(
+			ctx,
+			resource.Dependencies{encoder.Name(): encoder},
+			conf,
+			logger,
+		)
+		test.That(t, err, test.ShouldBeNil)
+		motorMap[name] = thisMotor
+	}
+	return motorMap
+}
+
+type noisyMovementSensor struct {
+	movementsensor.MovementSensor
+	mu         sync.Mutex
+	noise      spatialmath.Pose
+	queryCount int // Apply noise every other iteration when this is > 1
+}
+
+func (m *noisyMovementSensor) Position(ctx context.Context, extra map[string]interface{}) (*geo.Point, float64, error) {
```

Comment:
How does movement-sensor know where is it now? Is this because we are using mocked encoders and wheeledOdometry?
Why aren't the encoders and wheeledOdometry informing the ms here?

On main (https://github.com/viamrobotics/rdk/blob/main/services/motion/builtin/builtin_utils_test.go#L178) we currently have the dyanimic movement sensor. 
Can you please explain a bit more about how we accomplished its removal?

Reply:
> Is this because we are using mocked encoders and wheeledOdometry?

That is correct.

The struct you are commenting on here is only used in a handful of tests to ensure position-based replanning triggers; it wraps the "real" mocked motion sensor used, which may be why it looks like there are pieces missing.

See below in `createMovementSensor` for the creation of the one used more broadly, which includes the creation of `wheeledOdometry` with encoded motors.